src/main/java/com/nike/cerberus/auth/connector/okta/OktaAuthHelper.java

```diff
@@ -0,0 +1,184 @@
+
+            throw ApiException.newBuilder()
+                    .withApiErrors(DefaultApiError.GENERIC_BAD_REQUEST)
+                    .withExceptionMessage(msg)
+                    .build();
+        }
+    }
+
+    /**
+     * Request for verifying a MFA factor.
+     *
+     * @param factorId MFA factor id
+     * @param stateToken State token
+     * @param passCode One Time Passcode from MFA factor
+     * @return Session login token
+     */
+    protected AuthResult verifyFactor(final String factorId,
+                                      final String stateToken,
+                                      final String passCode) {
+
+        final AuthResult authResult;
+        try {
+            authResult = this.authClient.authenticateWithFactor(stateToken, factorId, passCode);
+        } catch (IOException ioe) {
+            final String msg = String.format("stateToken: %s failed to verify 2nd factor for reason: %s",
+                    stateToken, ioe.getMessage());
+
+            throw ApiException.newBuilder()
+                    .withApiErrors(DefaultApiError.AUTH_BAD_CREDENTIALS)
+                    .withExceptionMessage(msg)
+                    .build();
+        }
+
+        return authResult;
+    }
+
+    /**
+     * Authenticate a user with Okta
+     *
+     * @param username   Okta username
+     * @param password   Okta password
+     * @param relayState Deep link to redirect user to after authentication
+     * @return Session login token
+     */
+    protected AuthResult authenticateUser(final String username, final String password,
+                                          final String relayState) {
+
+        try {
+            return this.authClient.authenticate(username, password, relayState);
+        } catch (IOException ioe) {
+            final String msg = String.format("failed to authenticate user for reason: %s", ioe.getMessage());
+
+            throw ApiException.newBuilder()
+                    .withApiErrors(DefaultApiError.GENERIC_BAD_REQUEST)
+                    .withExceptionMessage(msg)
+                    .build();        }
+    }
+
+    /**
+     * Convenience method for parsing the Okta response and mapping it to a class.
+     *
+     * @param authResult  The Okta authentication result object
+     * @return Deserialized object from the response body
+     */
+    protected EmbeddedAuthResponseDataV1 getEmbeddedAuthData(final AuthResult authResult) {
+
+        Preconditions.checkArgument(authResult != null, "auth result cannot be null.");
+
+        final Map<String, Object> embedded = authResult.getEmbedded();
+
+        try {
+            final String embeddedJson = objectMapper.writeValueAsString(embedded);
+            return objectMapper.readValue(embeddedJson, EmbeddedAuthResponseDataV1.class);
+        } catch (IOException e) {
+            throw ApiException.newBuilder()
+                    .withApiErrors(DefaultApiError.SERVICE_UNAVAILABLE)
```

Comment:
Why would you want to return a 503 here?
It seems like this should be a 500 based error.